### PR TITLE
Update template fan for new entity model

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Update template fan for new entity model

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/45478
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by bdraco** [comment]
Marking as draft until the `preset_mode` concern is addressed

File: source/_integrations/fan.template.markdown [hunk 1]
```diff
@@ -26,17 +26,22 @@ fan:
...
-          service: script.fan_speed
+        set_percentage:
+          service: script.fans_set_speed
           data:
-            speed: "{{ speed }}"
+            percentage: "{{ percentage }}"
+        set_preset_mode:
+          service: script.fans_set_preset_mode
+          data:
+            preset_mode: "{{ preset_mode }}"
```
Comment (frenck) [inline source/_integrations/fan.template.markdown]:
```suggestion
            preset_mode: "{{ preset_mode }}"
```

File: source/_integrations/fan.template.markdown [hunk 2]
```diff
@@ -108,9 +117,9 @@ fan:
...
         required: false
         type: action
-      speeds:
-        description: List of speeds the fan is capable of running at.
+      preset_modes:
+        description: List of preset modes the fan is capable of.
         required: false
         type: [string, list]
-        default: ['low', 'medium', 'high']
+        default: []
```
Comment (frenck) [inline source/_integrations/fan.template.markdown]:
```suggestion
        default: []
```

File: source/_integrations/fan.template.markdown [hunk 3]
```diff
@@ -71,10 +76,14 @@ fan:
...
-      speed_template:
-        description: Defines a template to get the speed of the fan.
+      percentage_template:
+        description: Defines a template to get the speed percentage of the fan.
         required: false
         type: template
+      preset_mode_template:
+        description: Defines a template to get the preset mode of the fan.
+        required: false
+        type: template
```
Comment (frenck) [inline source/_integrations/fan.template.markdown]:
```suggestion
        type: template
```

**Review by frenck — APPROVED** [review]

**Pull request merged**